core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -5238,6 +5238,23 @@ private ImmutableList<ImmutableBitSet> cube(ImmutableBitSet... sets) {
         .fails("Type mismatch in column 1 of UNION");
 
     sql("select ^slacker^ from emp union select name from dept").ok();
+
+    sql("select ^name^ from dept union select name from dept union select slacker from emp")
```

Comment:
I am assuming these were failing with an exception previously.

Reply:
exactly, it was failing  with runtime exception from `Util#needToImplement`